cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java

```diff
@@ -855,4 +815,61 @@ private void toAuthorizables(EntityId entityId, List<org.apache.sentry.core.comm
         throw new IllegalArgumentException(String.format("The entity %s is of unknown type %s", entityId, entityType));
     }
   }
+
+  Set<ActionFactory.Action> toSentryActions(Set<Action> actions) {
+    Set<ActionFactory.Action> sentryActions = new HashSet<>(actions.size());
+    for (Action action : actions) {
+      sentryActions.add(new ActionFactory.Action(action.name()));
+    }
+    return Collections.unmodifiableSet(sentryActions);
+  }
+
+  private static List<Authorizable> toAuthorizables(List<TAuthorizable> tAuthorizables) {
+    List<Authorizable> authorizables = new ArrayList<>(tAuthorizables.size());
+    for (TAuthorizable tAuthorizable : tAuthorizables) {
+      authorizables.add(ModelAuthorizables.from(tAuthorizable.getType(), tAuthorizable.getName()));
+    }
+    return authorizables;
+  }
+
+  private Set<String> fetchGroups(Principal principal) {
+    return authProvider.getGroupMapping().getGroups(principal.getName());
+  }
+
+  private Set<Role> fetchRoles(final String group) throws Exception {
+    Set<TSentryRole> tSentryRoles = execute(new Command<Set<TSentryRole>>() {
+      @Override
+      public Set<TSentryRole> run(SentryGenericServiceClient client) throws Exception {
+        return client.listRolesByGroupName(sentryAdminGroup, group, COMPONENT_NAME);
+      }
+    });
+    Set<Role> roles = new HashSet<>();
+    for (TSentryRole tSentryRole : tSentryRoles) {
+      roles.add(new Role(tSentryRole.getRoleName()));
+    }
+    return roles;
+  }
+
+  private Set<WildcardPolicy> fetchPolicies(final Role role) throws Exception {
+    Set<TSentryPrivilege> sentryPrivileges = execute(new Command<Set<TSentryPrivilege>>() {
+      @Override
+      public Set<TSentryPrivilege> run(SentryGenericServiceClient client) throws Exception {
+        return client.listPrivilegesByRoleName(sentryAdminGroup, role.getName(), COMPONENT_NAME, instanceName);
+      }
+    });
+
+    if (sentryPrivileges == null) {
```

Comment:
From what I understand `SentryGenericServiceClient.listPrivilegesByRoleName()` can return null. Any particular reason you say this check is not needed?